### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by gemmen29**
## In this project we did the following:
- [x] #1
- [x] #2
- [x] #3
- [x] #4
- [x] #5
- [x] #6
- [x] #7
- [x] #8
- [x] #9
- [x] #10
- [x] #11
- [x] #12
- [x] #13
- [x] #14
- [x] #15

### CODE REVIEW

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi Team :clap:  bravo :100: 

## Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  <div>
+    <%= form.label :description, style: "display: block" %>
+    <%= form.text_area :description %>
+  </div>
+
+  <div>
+    <%= form.label :public, style: "display: block" %>
+    <%= form.check_box :public %>
+  </div>
```
[the remark this answers is not in the document]
Comment (TheLuisAbarca) [reply app/views/recipes/_form.html.erb]: Thanks for your indications and feedback @andersonlebon! :+1: :1st_place_medal:  Your suggestions were implemented on commit 5b56842.

File: app/models/food.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Food < ApplicationRecord
+  belongs_to :user
+  has_many :recipe_foods
+end
```
[the remark this answers is not in the document]
Comment (TheLuisAbarca) [reply app/models/food.rb]: Thanks for your indications and feedback @andersonlebon! :+1: :1st_place_medal:  Your suggestions were implemented on commit 5b56842.

File: app/controllers/foods_controller.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+class FoodsController < ApplicationController
+  before_action :set_food, only: %i[show update destroy]
+
+  # GET /foods or /foods.json
+  def index
+    @foods = Food.order(:id).page(params[:page]).per(2)
+  end
```
[the remark this answers is not in the document]
Comment (TheLuisAbarca) [reply app/controllers/foods_controller.rb]: Thanks for your indications and feedback @andersonlebon! :+1: :1st_place_medal:  Your suggestions were implemented on commit 5b56842.

File: app/views/recipe_foods/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  <% end %>
+
+  <div>
+    <%= form.label :food_id, style: "display: block" %>
+    <%= form.select :food_id, @foods.collect { |food| [food.name, food.id] } %>
+  </div>
+  
+  <div>
+    <%= form.label :quantity, style: "display: block" %>
+    <%= form.number_field :quantity %>
```
[the remark this answers is not in the document]
Comment (TheLuisAbarca) [reply app/views/recipe_foods/edit.html.erb]: Thanks for your indications and feedback @andersonlebon! :+1: :1st_place_medal:  Your suggestions were implemented on commit 5b56842.

File: app/views/foods/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    <% else %>
+      <% @foods.each do |food| %>
+        <tr class="text-center">
+          <td><%= food.name %></td>
+          <td><%= food.measurement_unit %></td>
+          <td><%= food.price %></td>
+          <!--td><%= link_to 'Delete', food_path(food.id) %></td-->
+          <% if current_user%>
+            <td><%= button_to "Delete", food, method: :delete, class: 'btn btn-danger'%></td>
+          <% end %>
```
[the remark this answers is not in the document]
Comment (TheLuisAbarca) [reply app/views/foods/index.html.erb]: Thanks for your indications and feedback @andersonlebon! :+1: :1st_place_medal:  Your suggestions were implemented on commit 5b56842.

File: app/views/recipes/_recipe.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<div class="container" id="<%= dom_id recipe %>">
+  <div class="row border border-dark border-2 p-2">
+    <div class="d-flex flex-column col-4">
+      <h2><%=recipe.name%></h2>
+      <% if current_user.id == recipe.user_id%>
+        <%= button_to "Remove", recipe, method: :delete, class: 'btn btn-danger'%>
+      <% end %>
+    </div>
```
[the remark this answers is not in the document]
Comment (TheLuisAbarca) [reply app/views/recipes/_recipe.html.erb:8]: Thanks for your indications and feedback @andersonlebon!  We are going to take into consideration your suggestion for further projects. :+1: :1st_place_medal:

File: app/views/foods/index.html.erb (hunk 5 above)
[the remark this answers is not in the document]
Comment (rloterh) [reply app/views/foods/index.html.erb]: This issue is well resolved! ✔️

File: app/controllers/foods_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (rloterh) [reply app/controllers/foods_controller.rb]: This issue is well resolved! ✔️

File: app/models/food.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (rloterh) [reply app/models/food.rb]: This issue is well resolved ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi  George & Luís, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comment(s) marked **[OPTIONAL]** under this review_
<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**